Neo4j.Driver/Neo4j.Driver/Internal/Session.cs

```diff
@@ -53,36 +54,65 @@ internal partial class Session : StatementRunner, ISession
         }
 
-        public override IStatementResult Run(Statement statement)
+        public IStatementResult Run(Statement statement, TransactionConfig txConfig)
         {
             return TryExecute(() =>
             {
                 EnsureCanRunMoreStatements();
                 _connection = _connectionProvider.Acquire(_defaultMode);
                 var protocol = _connection.BoltProtocol;
-                return protocol.RunInAutoCommitTransaction( _connection, statement, this);
+                return protocol.RunInAutoCommitTransaction( _connection, statement, this, _bookmark, txConfig);
             });
         }
 
-        public override Task<IStatementResultCursor> RunAsync(Statement statement)
+        public Task<IStatementResultCursor> RunAsync(Statement statement, TransactionConfig txConfig)
         {
             return TryExecuteAsync(async () =>
             {
                 await EnsureCanRunMoreStatementsAsync().ConfigureAwait(false);
                 _connection = await _connectionProvider.AcquireAsync(_defaultMode).ConfigureAwait(false);
                 var protocol = _connection.BoltProtocol;
-                return await protocol.RunInAutoCommitTransactionAsync(_connection, statement, this).ConfigureAwait(false);
+                return await protocol.RunInAutoCommitTransactionAsync(_connection, statement, this, _bookmark, txConfig).ConfigureAwait(false);
             });
         }
+        
+        public IStatementResult Run(string statement, TransactionConfig txConfig)
+        {
+            return Run(new Statement(statement), txConfig);
+        }
 
-        public ITransaction BeginTransaction()
+        public Task<IStatementResultCursor> RunAsync(string statement, TransactionConfig txConfig)
         {
-            return TryExecute(() => BeginTransactionWithoutLogging(_defaultMode));
+            return RunAsync(new Statement(statement), txConfig);
         }
 
-        public Task<ITransaction> BeginTransactionAsync()
+        public IStatementResult Run(string statement, IDictionary<string, object> parameters, TransactionConfig txConfig)
         {
-            return TryExecuteAsync(() => BeginTransactionWithoutLoggingAsync(_defaultMode));
+            return Run(new Statement(statement, parameters), txConfig);
+        }
+
+        public Task<IStatementResultCursor> RunAsync(string statement, IDictionary<string, object> parameters, TransactionConfig txConfig)
+        {
+            return RunAsync(new Statement(statement, parameters), txConfig);
+        }
+        
+        public override IStatementResult Run(Statement statement)
+        {
+            return Run(statement, TransactionConfig.Empty);
+        }
+        public override Task<IStatementResultCursor> RunAsync(Statement statement)
+        {
+            return RunAsync(statement, TransactionConfig.Empty);
+        }
+
+        public ITransaction BeginTransaction(TransactionConfig txConfig=null)
```

Comment:
Shall we extract method overloads instead of default parameters (for better C++/CLI integration) in the rest of the class (for `public` methods)?